src/event/log_event.rs

```diff
@@ -135,32 +162,81 @@ impl LogEvent {
         trace!(entry = ?current_pointer, "Result.");
         Ok(current_pointer)
     }
+
+    /// Merge all fields specified at `fields` from `incoming` to `current`.
+    pub fn merge(&mut self, mut incoming: LogEvent, fields: &[impl AsRef<str>]) {
+        for field in fields {
+            let incoming_val = match incoming.remove(field) {
+                None => continue,
+                Some(val) => val,
+            };
+            match self.get_mut(&field) {
+                None => {
+                    self.insert(field, incoming_val);
+                }
+                Some(current_val) => current_val.merge(incoming_val),
+            }
+        }
+        self.metadata.merge(incoming.metadata());
+    }
+}
+
+impl Equivalent for LogEvent {
```

Comment:
The goal was to provide a parallel to `assert_eq` where differences unrelated to the core content of the event were not compared. Right now this excludes the timestamp metadata, but eventually it may include other metadata.

The proposed alternative is not a good solution. There are times where we do want to compare events being *exactly* identical in tests (ie ensuring metadata is preserved in a transform) and times where we do not (ie events coming out of a source, where only the timestamp is unpredictable), so there has to be a comparison separate from `Eq`.

While implementing this, there were a couple of places where `Result`s were compared this way, so the trait path was a natural fit.

Reply:
Gotcha, yeah, I see. Something still seems off to me about it, but I can't place it. I think that it is that I'm struggling to come up with a succinct definition for what makes two types `Equivalent` like I can for `Eq` and `PartialEq`.

Reply:
Maybe a different name would help, something in the line of `EqContent`

Reply:
I am definitely open to an alternate naming for this trait. @jszwedko does `EqContent` make more sense to you, or would another name make more sense?

Reply:
`EqContent` is an improvement (I'd probably write it as `ContentEq` to match `PartialEq`). I'm still not crazy about that one either, but I'm struggling to come up with a better one.

Maybe something specific like `EventDataEq`?

Reply:
`EventEq` sounds good to me since this trait is specific for `Event` and we are comparing two events for equality that just so happen to contain reference to metadata.

Reply:
While it is specific to `Event` (in that it only differs from `eq` for `Event`), it does not compare the whole event, so I think I prefer `EventDataEq`. Of course, this might make the assertion macro ugly, but so be it.

Reply:
👍 I'll take ugly but clear 😄